conan/api/subapi/export.py

```diff
@@ -1,27 +1,111 @@
+import os
+from typing import List, Tuple
+
+from conan import ConanFile
 from conan.api.output import ConanOutput
+from conan.cli.printers.graph import print_graph_basic
 from conan.internal.cache.cache import PkgCache
 from conan.internal.conan_app import ConanApp
 from conan.internal.api.export import cmd_export
 from conan.internal.methods import run_package_method
 from conan.internal.graph.graph import BINARY_BUILD, RECIPE_INCACHE
-from conan.api.model import PkgReference
+from conan.api.model import PkgReference, Remote, RecipeReference
 from conan.internal.util.files import mkdir
 
 
 class ExportAPI:
+    """ This API provides methods to export artifacts, both recipes and pre-compiled package
+    binaries from user folders to the Conan cache, as Conan recipes and Conan package binaries
+    """
 
     def __init__(self, conan_api, helpers):
         self._conan_api = conan_api
         self._helpers = helpers
 
-    def export(self, path, name, version, user, channel, lockfile=None, remotes=None):
+    def export(self, path, name: str = None, version: str = None, user: str = None,
+               channel: str = None, lockfile=None,
+               remotes: List[Remote] = None) -> Tuple[RecipeReference, ConanFile]:
+        """
+        Exports a conanfile.py recipe, together with its associated files to the Conan cache.
+        A "recipe-revision" will be computed and assigned.
+        :param path: Path to the conanfile to be exported
+        :param name: Optional package name. Typically not necessary as it is defined by the recipe
```

Comment:
Maybe we should add in this ref args that if the recipe-defined and the passed arguments mismatch, an exception will be raised?